include/boost/geometry/views/detail/boundary_view/implementation.hpp

```diff
@@ -0,0 +1,400 @@
+        typename OtherValue,
+        typename OtherReference,
+        typename OtherDifference
+    >
+    friend class polygon_rings_iterator;
+
+
+    static inline std::size_t num_rings(Polygon const& polygon)
+    {
+        return geometry::num_interior_rings(polygon) + 1;
+    }
+
+    inline Reference dereference() const
+    {
+        if (m_index == 0)
+        {
+            return exterior_ring(*m_polygon);
+        }
+        return range::at(interior_rings(*m_polygon), m_index - 1);
+    }
+
+    template
+    <
+        typename OtherPolygon,
+        typename OtherValue,
+        typename OtherReference,
+        typename OtherDifference
+    >
+    inline bool equal(polygon_rings_iterator
+                      <
+                          OtherPolygon,
+                          OtherValue,
+                          OtherReference,
+                          OtherDifference
+                      > const& other) const
+    {
+        return m_polygon == other.m_polygon && m_index == other.m_index;
+    }
+
+    inline void increment()
+    {
+        ++m_index;
+    }
+
+    inline void decrement()
+    {
+        --m_index;
+    }
+
+    template
+    <
+        typename OtherPolygon,
+        typename OtherValue,
+        typename OtherReference,
+        typename OtherDifference
+    >
+    inline Difference distance_to(polygon_rings_iterator
+                                  <
+                                      OtherPolygon,
+                                      OtherValue,
+                                      OtherReference,
+                                      OtherDifference
+                                  > const& other) const
+    {
+        return other.m_index - m_index;
+    }
+
+    inline void advance(Difference n)
+    {
+        m_index += n;
+    }
+
+private:
+    Polygon* m_polygon;
+    std::size_t m_index;
+};
+
+
+template <typename Ring>
+class ring_boundary : closeable_view<Ring, closure<Ring>::value>::type
+{
+private:
+    typedef typename closeable_view<Ring, closure<Ring>::value>::type base_type;
+
+public:
+    typedef typename base_type::iterator iterator;
+    typedef typename base_type::const_iterator const_iterator;
+    
+    typedef linestring_tag tag_type;
+
+    explicit ring_boundary(Ring& ring)
+        : base_type(ring) {}
+
+    iterator begin() { return base_type::begin(); }
+    iterator end() { return base_type::end(); }
+    const_iterator begin() const { return base_type::begin(); }
+    const_iterator end() const { return base_type::end(); }
+};
+
+
+template <typename Polygon>
+class polygon_boundary
+{
+    typedef boundary_view<typename ring_type<Polygon>::type> boundary_view_type;
+    typedef polygon_rings_iterator<Polygon> rings_iterator_type;
+
+    inline void initialize_views(Polygon& polygon)
+    {
+        std::pair<boundary_view_type*, std::ptrdiff_t> result
+            = std::get_temporary_buffer<boundary_view_type>(m_num_rings);
```

Comment:
I just checked how exactly this function works and it seems that AFAIU it can allocate less elements than it's requested. This means that if not enough space was allocated, the constructors would be called for unallocated memory. You could e.g. replace the condition below with this:

    result.second < m_num_rings

or just call `new` (and then get rid of the `bad_alloc` throw):

    m_views = new char[sizeof(boundary_view_type) * m_num_rings];

Then also m_num_rings could be unsigned, e.g. `std::size_t`. It could be `range_size<[polygon interior rings range]>::type` but since it's used with `new` it probably doesn't have sense.